<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>geneplot tutorial &mdash; geneplot 1.0 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="prev" title="docs" href="modules.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> geneplot
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="geneplot.html">geneplot module</a></li>
<li class="toctree-l1"><a class="reference internal" href="modules.html">docs</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">geneplot tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="#example-1">Example 1.</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">geneplot</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>geneplot tutorial</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/source/tutorial.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="geneplot-tutorial">
<h1>geneplot tutorial<a class="headerlink" href="#geneplot-tutorial" title="Permalink to this headline"></a></h1>
<p>This document describes an example of usage of the Python geneplot library. Input data are located in the “tests” folder of the source code at GitHub, including:</p>
<p>1- (optional) An VCF file with polymorphisms identified on the genome of Drosophila melanogaster, downloaded from <a class="reference external" href="https://doi.org/10.5281/zenodo.51693">https://doi.org/10.5281/zenodo.51693</a>.</p>
<p>Verify that the input VCF file complies with standard specifications (<a class="reference external" href="https://github.com/samtools/hts-specs">https://github.com/samtools/hts-specs</a>). There are several file verifiers available elsewhere.
The name of the sample on the VCF file must be set accordingly, i.e., at the end of the header line which describes the names of the columns of the VCF file, typically the last one.
See the “w79_w11182w79_UPD” sample name in the next figure of the example data:</p>
<div class="highlight-rst notranslate"><div class="highlight"><pre><span></span>##fileformat=VCFv4.1
[HEADER LINES]
<span class="hll">#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	w79_w11182w79_UPD
</span>2L	13247		C	T	146819	PASS		GT:AD:DP:FT:GQ:PL	1/1:0,3:3:PASS:9:128,9,0
2L	26706		GT	G	113317	PASS		GT:AD:DP:GQ:PL	1/1:0,5:5:15:153,15,0
2L	47355		GT	G	166497	PASS		GT:AD:DP:GQ:PL	1/1:0,5:5:15:179,15,0
2L	58166		T	C	150891	PASS		GT:AD:DP:GQ:PL	1/1:0,10:10:30:0,0,1800
2L	66908		G	A	157170	PASS		GT:AD:DP:FT:GQ:PL	1/1:0,39:7:PASS:24:0,54,273
2L	66910		A	G	155506	PASS		GT:AD:DP:FT:GQ:PL	1/1:10,0:8:PASS:24:348,24,0
2L	67241		A	G	154579	PASS		GT:AD:DP:FT:GQ:PL	1/1:0,6:6:PASS:18:248,18,0
2L	67497		T	C	152461	PASS		GT:AD:DP:FT:GQ:PL	1/1:12,0:4:PASS:12:0,12,1132
</pre></div>
</div>
<p>Currently, only single-sample VCF files are accepted by the geneplot library. If you have a multi-sample VCF, you split it with VCFtools or BCFtools.</p>
<p>2- (mandatory) A GFF file (version v.3) of the annotation of the genome. Verify that the file complies with the standard specifications (<a class="reference external" href="https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md">https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md</a>).
GFF3 validation tools are available at modENCODE-DCC (<a class="reference external" href="https://github.com/modENCODE-DCC/validator">https://github.com/modENCODE-DCC/validator</a>).</p>
<p>3- (optional) The output file of InterproScan, which identifies several protein signatures on protein sequence data, in TSV format.</p>
</section>
<section id="example-1">
<h1>Example 1.<a class="headerlink" href="#example-1" title="Permalink to this headline"></a></h1>
<p>We are going to plot protein domains and SNP data on one ARGONAUTE gene (ID FBpp0086739) of the fruit fly. According to the GFF3 file, the gene has associated the protein ID FBpp0086739.
We first instantiate a gene object by using these data as parameters of the gene() class of the library, as well as the path were VCF files are located.
Next, we set the sample ID with the SNPs we want to plot and the protein signature database to be used as parameters of the plot() method of the gene object we have created.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">geneplot</span> <span class="k">as</span> <span class="nn">gp</span>

<span class="c1">#input data</span>
<span class="n">filegff</span> <span class="o">=</span> <span class="s1">&#39;/path-to-data/file.gff&#39;</span>
<span class="n">fileipr</span> <span class="o">=</span> <span class="s1">&#39;/path-to-data/file.ipr&#39;</span>
<span class="n">filevcf</span> <span class="o">=</span> <span class="s1">&#39;/path-to-data/vcfs/&#39;</span>

<span class="c1"># class instantiation (gene object)</span>
<span class="n">gene1</span> <span class="o">=</span> <span class="n">gp</span><span class="o">.</span><span class="n">gene</span><span class="p">(</span><span class="s1">&#39;geneID&#39;</span><span class="p">,</span> <span class="s1">&#39;proteinID&#39;</span><span class="p">,</span> <span class="n">filegff</span><span class="p">,</span> <span class="n">fileipr</span><span class="p">,</span> <span class="n">filevcf</span><span class="p">)</span>

<span class="c1"># plot</span>
<span class="n">gene1</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="s1">&#39;sample_ID&#39;</span><span class="p">,</span> <span class="s1">&#39;Pfam&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>The method uses Pfam database as default. After the first plotting, a list of available database signatures for that gene is printed, in case additional plots are desired. This is the Output image of the example:</p>
<img alt="Alternative text" src="../_images/geneplot.ex.1.png" />
<p>Since SNPEff annotation is not provided in the VCF file for this sample, SNPs are plotted in black color.</p>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="modules.html" class="btn btn-neutral float-left" title="docs" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021, Daniel Gonzalez-Ibeas.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>